mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -113,11 +137,15 @@ where
         block_offset: u64,
         object_size: usize,
     ) -> DataCacheResult<Option<ChecksummedBytes>> {
```

Comment:
Too many points of failure.
Shall we factor out the logic into an inner method and only handle the metrics here?

Reply:
I'm not planning on refactoring into an inner method, as that makes the reasons harder to propagate

Reply:
An alternative could be to still split into an inner function which records the failure (and reason), but move the hit/miss metric to the wrapper.

Happy to review the approach later, though.